vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/validators/all_output_values_not_empty_validator.tsx

```diff
@@ -0,0 +1,16 @@
+import { inputValueNonEmpty } from '../../../../../selectors/workflow_selectors';
+import {DataEnvelope, ValidationInputSpecification} from '../input_types';
+import AllInnerValuesNotEmptyValidator, { AllInnerValuesNotEmptyValidatorStrong } from './all_inner_values_not_empty_validator';
+import NotEmptyValidator from './not_empty_validator';
+
+const AllOutputValuesNotEmptyValidator = (input: ValidationInputSpecification<DataEnvelope<any>, DataEnvelope<any>>): string[] => {
+  // When this is used, we don't care if keys in input.data are missing from input.value
+  if (!inputValueNonEmpty(input.value)) {
+    return ['Value cannot be empty.'];
+  } else {
+    input.data = input.value as DataEnvelope<any>
```

Comment:
This is fairly dangerous.  You are mutating the passed input reference, which will effect not just your local variable _but also magically all references to the passed input from other code_.  I believe what you want is this:

```
AllOutputValuesNotEmptyValidator({...input, data: input.value})
```

This syntax essentially creates a shallow copy of input, such that the data key is replaced with input.value (but not mutated).
```